pkg/resource/deploy/source_eval.go

```diff
@@ -1424,6 +1434,24 @@ func (rm *resmon) RegisterResource(ctx context.Context,
 	}
 
 	additionalSecretOutputs := req.GetAdditionalSecretOutputs()
+	if len(additionalSecretOutputs) == 0 && rm.loader != nil {
+		// If the request did not specify additional secret outputs,
+		// we'll try to load the package reference and infer which properties are
+		// supposed to be secret.
```

Comment:
This should be unconditional. Even in the future when SDK gen doesn't generating anything here users are free to add to this list via the resource option. The schema info should be added to that.

Reply:
Done ✅